src/epanet.c

```diff
@@ -3261,7 +3261,7 @@ int DLLEXPORT EN_deletelink(EN_Project p, int index, int actionCode)
 
     // Remove link's comment
     free(net->Link[index].Comment);
-    net->Link[net->Nlinks].Comment = NULL;
```

Comment:
I think the intent of this line was to nullify the now-free'd pointer. If I'm right, it should have read:
```
net->Link[index].Comment = NULL;
```

Reply:
I don't think it's necessary to set the `Comment `pointer of the deleted link to `NULL `since it will be assigned to the `Comment `pointer of the next higher link when the link array's entries are shifted down by one. However, the original line of code that was deleted should remain, but be moved to right after the array shifting is done, so as not to leave the `Comment `pointer of the now unused last entry of the array invalid.

A similar statement that sets the last node's `Comment `to `NULL `when a node is deleted appears in the proper location (after the nodes are shifted) in `EN_deletenode()` (see line 1898 of `epanet.c`). The EN_deletepattern() and EN_deletecurve() are missing the statements that set the pointers of the remaining unused array object to NULL after shifting occurs.

Having said all this I'm not sure that it's really necessary to `NULL `the pointers in these last unused array entries since they will be assigned to `NULL `when a new object is added via an `EN_addxxx()` function and will not tried to be free'd when the project is closed since that action is only taken for the array entries below them.

Reply:
I also think not needed. I have removed the lines and from `EN_deletenode` and it looks ok.